### ISSUE TITLE
Java: using addAllowedAssignments causes JVM crash after Solver tries to solve model

### ISSUE THREAD

**Issue by Dignitas** [google/or-tools#1015]
With the code example below I'm able to reproduce the crash, which root cause is calling the `addAllowedAssignments()`. When using the `addForbiddenAssignments()` instead this crash does no longer occur.

```java
private static IntVar[] entitiesOne;
static {
	System.loadLibrary("jniortools");
}
public static void main(final String[] args) {
	run();
}

private static void run() {
	final CpModel model = new CpModel();
	final int numEntityOne = 50000;
	final int numEntityTwo = 100;
	entitiesOne = new IntVar[numEntityOne];
	for (int i = 0; i < entitiesOne.length; i++) {
		entitiesOne[i] = model.newIntVar(1, numEntityTwo, "E" + i);
	}
	final int[][] allAllowedValues = new int[numEntityTwo][entitiesOne.length];
	for (int i = 0; i < numEntityTwo; i++) {
		for (int j = 0; j < entitiesOne.length; j++) {
			allAllowedValues[i][j] = i;
		}
	}
	try {
		model.addAllowedAssignments(entitiesOne, allAllowedValues);
	} catch (final Exception e) {
		e.printStackTrace();
	}
		final Random r = new Random();
	for (int i = 0; i < entitiesOne.length; i++) {
		model.addEquality(entitiesOne[i], r.nextInt((numEntityTwo)));
	}
		final CpSolver solver = new CpSolver();
	solver.solve(model);
}
```

**Comment by lperron** [comment]
Just tested in the source, It works fine.

**Comment by raphweb** [comment]
Bug was in 6.9.1, apparently already fixed with 6.10.
